### PR TITLE
Optional metrics reporting via statsd

### PR DESCRIPTION
**Pull request by wvj384**
Parameter `metrics_method` added to application config with the following possible values:
`log` - current behaviour (default value), send metrics via log messages
`statsd` - start statsd client in `erllambda_config_srv` init and send metrics via it

### CODE REVIEW

File: src/erllambda.erl [hunk 1]
```diff
@@ -406,6 +388,43 @@ reformat([], _Width) ->
 one_line_it(Text) ->
     re:replace(string:trim(Text), "\r?\n\s*", " ", [{return,list},global,unicode]).
 
+%% Sends metric via log message from lambda using following format
+%% MONITORING|unix_epoch_timestamp|metric_value|metric_type|my.metric.name|#tag1:value,tag2
+%% where metric_type :: "count" | "gauge" | "histogram"
+metric_by_log(MName, Val, Type, Tags) ->
```
Comment (motobob) [inline src/erllambda.erl:394]: metric_via_log

File: test/erllambda_handler_crash_SUITE.erl [hunk 2]
```diff
@@ -123,8 +123,8 @@ test_unhandled_stacktrace(_Config) ->
     {error, #{<<"stackTrace">> := Trace}} =
         erllambda_aws_runtime:call(#{type => badmatch}),
     ["{erllambda_crash_handler,handle,2,[{file,\"file.erl\"},{line,11}]}",
-     "{erllambda,invoke_exec,3,[{file,\"file.erl\"},{line,305}]}",
-     "{erllambda,'-invoke/3-fun-0-',4,[{file,\"file.erl\"},{line,285}]}"] =
+     "{erllambda,invoke_exec,3,[{file,\"file.erl\"},{line,287}]}",
```
Comment (motobob) [inline test/erllambda_handler_crash_SUITE.erl:126]: i do not remember why this file is even here)

File: test/erllambda_ct.erl [hunk 3]
```diff
@@ -56,6 +56,9 @@ record_to_map(R) when is_record(R, aws_config) ->
     record_to_map(R, Ks);
 record_to_map(R) when is_record(R, aws_assume_role) ->
     Ks = record_info(fields, aws_assume_role),
+    record_to_map(R, Ks);
+record_to_map(R) when is_record(R, hackney_client_options) ->
+    Ks = record_info(fields, hackney_client_options),
```
Comment (motobob) [inline test/erllambda_ct.erl:61]: why is this here?

File: rebar.lock [hunk 4]
```diff
@@ -1,16 +1,24 @@
 {"1.1.0",
 [{<<"base16">>,{pkg,<<"base16">>,<<"1.0.0">>},1},
+ {<<"dogstatsd">>,
```
Comment (motobob) [inline rebar.lock:3]: we should probably kill the lock file at the library/app level here.

**Review by motobob — CHANGES_REQUESTED** [review]
is there a "flush" API of the statsD app here as we might need it.

File: test/erllambda_ct.erl (hunk 3 above)
Comment (wvj384) [reply test/erllambda_ct.erl:61]: tests failed on my machine without such clause...

**Comment by wvj384** [comment]
> is there a "flush" API of the statsD app here as we might need it.

No, as I understand flush is controlled by the DD extension

File: rebar.lock (hunk 4 above)
Comment (wvj384) [reply rebar.lock:3]: ok, let's do this

**Review by motobob — APPROVED** [review]

**Comment by wvj384** [comment]
@motobob Made a fix for count metrics, described the idea in the comment. Link to DD docs https://docs.datadoghq.com/metrics/types/?tab=rate#submission-types-and-datadog-in-app-types

File: test/erllambda_ct.erl (hunk 3 above)
Comment (motobob) [reply test/erllambda_ct.erl:61]: well let's find out why and fix. this should not be here.

**Review by motobob — CHANGES_REQUESTED** [review]
last comment please

**Review by motobob — APPROVED** [review]
